src/document.js

```diff
@@ -690,7 +697,7 @@ class DocumentSnapshot {
           target[key] = {};
           merge(target[key], value, path, pos + 1);
         }
-      } else if (is.object(target[key])) {
+      } else if (isPlainObject(target[key])) {
```

Comment:
This can also be the an object that is passed in by the user if we are still at the root level.

Reply:
At the root level won't target be the {} object we create on line 712?  

I think the only way this will be a custom object is if you call expandMap({'foo': new Date(), 'foo.bar': 42}), which I think is invalid.  Oh!  But maybe that is what this code is actually trying to detect?  Gah. If so, I missed that. I thought 'data' was already validated.